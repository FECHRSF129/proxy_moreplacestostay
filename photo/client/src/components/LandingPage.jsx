import React from 'react';
import Description from './Description.jsx';
import MainPhotos from './MainPhotos.jsx';

function LandingPage(props) {
  return (
    <div>
      <Description data={props.data} share={props.share}/>
      <MainPhotos photos={props.photos} click={props.click}/>
    </div>
  )
}

export default LandingPage;