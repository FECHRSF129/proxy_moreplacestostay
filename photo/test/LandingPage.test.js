import LandingPage from '../client/src/components/LandingPage.jsx';
import Enzyme, { shallow, render, mount } from 'enzyme';
import 'jest-styled-components';

const wrapper = mount(<LandingPage data={
    [{
      Description: "Sumptuous estate with views of Hollywood",
      Location: "Hollywood, California, United States",
      Rating: 4.74,
      Superhost: 1,
      TotalReviews: 14,
      id: 65,
    }]
  }
    photos={
    [
      {
        PhotoURL: "https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/4bb58786-cf86-48ac-9203-cde18dd33b7e.jpg",
        PropertyID: 65,
        RoomID: 449,
        ShowOrder: 1,
        id: 1345
      },
      {
        PhotoURL: "https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/50ff73e3-9d7a-44f0-8014-c3eaf95e6029.jpg",
        PropertyID: 65,
        RoomID: 449,
        ShowOrder: 2,
        id: 1346
      },
      {
        PhotoURL: "https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/5e01ea9c-778d-4e44-823f-b0d55b9719ab.jpg",
        PropertyID: 65,
        RoomID: 449,
        ShowOrder: 3,
        id: 1347
      }
    ]
  }/>);

describe('LandingPage Test Suite', () => {

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render correctly with props', () => {
    expect(wrapper).toMatchSnapshot();
  });

});