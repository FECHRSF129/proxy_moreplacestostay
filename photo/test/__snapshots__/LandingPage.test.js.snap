// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Test Suite should render correctly with props 1`] = `
<LandingPage
  data={
    Array [
      Object {
        "Description": "Sumptuous estate with views of Hollywood",
        "Location": "Hollywood, California, United States",
        "Rating": 4.74,
        "Superhost": 1,
        "TotalReviews": 14,
        "id": 65,
      },
    ]
  }
  photos={
    Array [
      Object {
        "PhotoURL": "https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/4bb58786-cf86-48ac-9203-cde18dd33b7e.jpg",
        "PropertyID": 65,
        "RoomID": 449,
        "ShowOrder": 1,
        "id": 1345,
      },
      Object {
        "PhotoURL": "https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/50ff73e3-9d7a-44f0-8014-c3eaf95e6029.jpg",
        "PropertyID": 65,
        "RoomID": 449,
        "ShowOrder": 2,
        "id": 1346,
      },
      Object {
        "PhotoURL": "https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/5e01ea9c-778d-4e44-823f-b0d55b9719ab.jpg",
        "PropertyID": 65,
        "RoomID": 449,
        "ShowOrder": 3,
        "id": 1347,
      },
    ]
  }
>
  <div>
    <Description
      data={
        Array [
          Object {
            "Description": "Sumptuous estate with views of Hollywood",
            "Location": "Hollywood, California, United States",
            "Rating": 4.74,
            "Superhost": 1,
            "TotalReviews": 14,
            "id": 65,
          },
        ]
      }
    >
      <styled.div>
        <div
          className="sc-AxjAm jGqiXW"
        >
          <styled.div>
            <div
              className="sc-AxirZ hdtZFD"
            >
              <styled.h1>
                <h1
                  className="sc-AxhCb iHfCoW"
                >
                  Sumptuous estate with views of Hollywood
                </h1>
              </styled.h1>
              <styled.div>
                <div
                  className="sc-AxiKw hsVmMj"
                >
                  <styled.span>
                    <span
                      className="sc-AxgMl ccJqUs"
                    >
                      <span>
                        â˜…
                      </span>
                      <span>
                        4.74
                      </span>
                      <span>
                        (14)
                      </span>
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-Axmtr hgBReG"
                    >
                      Superhost
                    </span>
                  </styled.span>
                  <styled.span>
                    <span
                      className="sc-AxheI bRpUtj"
                    >
                      Hollywood, California, United States
                    </span>
                  </styled.span>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </Description>
    <MainPhotos
      photos={
        Array [
          Object {
            "PhotoURL": "https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/4bb58786-cf86-48ac-9203-cde18dd33b7e.jpg",
            "PropertyID": 65,
            "RoomID": 449,
            "ShowOrder": 1,
            "id": 1345,
          },
          Object {
            "PhotoURL": "https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/50ff73e3-9d7a-44f0-8014-c3eaf95e6029.jpg",
            "PropertyID": 65,
            "RoomID": 449,
            "ShowOrder": 2,
            "id": 1346,
          },
          Object {
            "PhotoURL": "https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/5e01ea9c-778d-4e44-823f-b0d55b9719ab.jpg",
            "PropertyID": 65,
            "RoomID": 449,
            "ShowOrder": 3,
            "id": 1347,
          },
        ]
      }
    >
      <styled.div>
        <div
          className="sc-AxmLO cFFrtF"
        >
          <styled.div>
            <div
              className="sc-fzozJi hbgEQd"
            >
              <img
                src="https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/4bb58786-cf86-48ac-9203-cde18dd33b7e.jpg"
              />
              <img
                src="https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/50ff73e3-9d7a-44f0-8014-c3eaf95e6029.jpg"
              />
              <img
                src="https://fec-airbnb-plus-photos.s3-us-west-1.amazonaws.com/5e01ea9c-778d-4e44-823f-b0d55b9719ab.jpg"
              />
            </div>
          </styled.div>
        </div>
      </styled.div>
    </MainPhotos>
  </div>
</LandingPage>
`;
